compiler/codegen/OMRCodeGenerator.hpp

```diff
@@ -616,8 +619,8 @@ class OMR_EXTENSIBLE CodeGenerator
    bool isMaterialized(TR::Node *); // no virt, cast
    bool isMaterialized(int64_t); // no virt, cast
    bool materializesLargeConstants() { return false;}
-   int32_t getLargestNegConstThatMustBeMaterialized() {return -32769;} // no virt, cast
-   int32_t getSmallestPosConstThatMustBeMaterialized() {return 32768;} // no virt, cast
+   int64_t getLargestNegConstThatMustBeMaterialized() {return -32769;}
+   int64_t getSmallestPosConstThatMustBeMaterialized() {return 32768;}
```

Comment:
Wouldn't it make more sense to do the extension point as Darryl suggested above and basically say no by default and anyone who wants it opts in. That way PPC specifically specifies what range they want and won't be broken by changes at the common codegen layer? It would seem to be more robust.